src/main/java/ch/njol/skript/bukkitutil/HealthUtils.java

```diff
@@ -95,21 +103,57 @@ public static void heal(Damageable e, double h) {
 		}
 		setHealth(e, getHealth(e) + h);
 	}
-	
+
 	public static double getDamage(EntityDamageEvent e) {
 		return e.getDamage() / 2;
 	}
-	
+
 	public static double getFinalDamage(EntityDamageEvent e) {
 		return e.getFinalDamage() / 2;
 	}
-	
+
 	public static void setDamage(EntityDamageEvent e, double damage) {
 		e.setDamage(damage * 2);
 	}
-	
-	public static void setDamageCause(Damageable e, DamageCause cause) {
-		e.setLastDamageCause(new EntityDamageEvent(e, cause, 0));
+
+	/**
+	 * In a late 1.20.4 version. Spigot added a non null DamageSource parameter in all EntityDamageEvent constructors,
+	 * and removed existing constructors that did not contain a DamageSource parameter.
+	 * @ScheduledForRemoval Existing until proper support is added to ExprLastDamageCause for DamageSource
+	 */
+	@Deprecated
+	@ScheduledForRemoval
+	public static boolean DAMAGE_SOURCE;
```

Comment:
Since this PR can go into feature, we should aim to include proper support in ExprLastDamageCause.

Reply:
It should go into it's own pull request. It's more complex then a few lines. You have to account for both DamageCause and DamageType in the same set expression.